lightning/src/offers/invoice.rs

```diff
@@ -491,19 +492,32 @@ where
 
 impl UnsignedBolt12Invoice {
 	fn new(invreq_bytes: &[u8], contents: InvoiceContents) -> Self {
+		const NON_EXPERIMENTAL_TYPES: core::ops::Range<u64> = 0..INVOICE_REQUEST_TYPES.end;
+		const EXPERIMENTAL_TYPES: core::ops::Range<u64> = 0..0;
+
+		let mut bytes = Vec::new();
```

Comment:
> I was thinking move `InvoiceContents` and store `invreq_bytes` as a `Vec`, rather than a slice (we currently always actually have an owned `Vec` at the callsite when calling here, afaict.

Hmm... `invreq_bytes` comes from `InvoiceBuilder` as a slice from either `&InvoiceRequest` or `&Refund`, so we don't own it. We could copy it, but that leaves us with a `Vec<u8>` containing a both experimental and non-experimental ranges for which we need to insert a signature.

> If we don't do this at a minimum we need a `with_capacity` here (overshooting a bit is fine, I think, this struct shouldn't be around long).

Done.